File: 30DaysOfCode/day25_runningTimeAndComplexity/runningTimeAndComplexity.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;


int main() {
    int n;
    cin >> n;
    
    for (int i = 0; i < n; i++) {
        int num;
        cin >> num;
        bool flag = true;
        
        if (num < 2) {
            cout << "Not prime" << endl;
        } else if (num <= 3) {
            cout << "Prime" << endl;
        } else if (num % 2 == 0 || num % 3 == 0) {
            cout << "Not prime" << endl;
        } else {
            for (int i = 5; i <= sqrt(num); i += 2) {
                if (num % i == 0) {
                    cout << "Not prime" << endl;
                    flag = false;
                    break;
                }
            }
            
            if (flag) {
                cout << "Prime" << endl;
            }
        }
    }
    
    return 0;
}

File: 30DaysOfCode/day25_runningTimeAndComplexity/runningTimeAndComplexity.js
function processData(input) {
    var inputArr = input.split('\n');
    
    for (var i = 1; i < inputArr.length; i++) {
        var num = inputArr[i];
        var flag = true;
        
        if (num < 2) {
            console.log('Not prime');
        } else if (num <= 3) {
            console.log('Prime');
        } else if (num % 2 == 0 || num % 3 == 0) {
            console.log('Not prime');
        } else {
            for (var j = 5; j <= Math.sqrt(num); j += 2) {
                if (num % j == 0) {
                    console.log('Not prime');
                    flag = false;
                    break;
                }
            }
            
            if (flag) {
                console.log('Prime');
            }
        }
    }
} 

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
   processData(_input);
});
